optuna/integration/mlflow.py

```diff
@@ -0,0 +1,141 @@
+import optuna
+from optuna import type_checking
+
+if type_checking.TYPE_CHECKING:
+    from typing import Dict  # NOQA
+
+try:
+    import mlflow
+
+    _available = True
+except ImportError as e:
+    _import_error = e
+    _available = False
+    mlflow = object
+
+
+def _check_mlflow_availability():
+    # type: () -> None
+
+    if not _available:
+        raise ImportError(
+            "MLflow is not available. Please install MLflow to use this "
+            "feature. It can be installed by executing `$ pip install "
+            "mlflow`. For further information, please refer to the installation guide "
+            "of MLflow. (The actual import error is as follows: " + str(_import_error) + ")"
+        )
+
+
+class MLflowCallback(object):
+    """Callback to track optuna trials with MLflow.
+
+    This callback adds relevant information that is
+    tracked by Optuna to MLflow.
+
+    Example:
+
+        Add MLflow callback to optuna optimization.
+
+        .. testsetup::
+
+            import pathlib
+            import tempfile
+            tempdir = tempfile.mkdtemp()
+            YOUR_TRACKING_URI = pathlib.Path(tempdir).as_uri()
+
+        .. testcode::
+
+            import optuna
+            from optuna.integration.mlflow import MLflowCallback
+
+            def objective(trial):
+                x = trial.suggest_uniform('x', -10, 10)
+                return (x - 2) ** 2
+
+            mlflc = MLflowCallback(
+                tracking_uri=YOUR_TRACKING_URI,
+                metric_name='my metric score',
+            )
+
+            study = optuna.create_study(study_name='my_study')
+            study.optimize(objective, n_trials=10, callbacks=[mlflc])
+
+        .. testcleanup::
+
+            import shutil
+            shutil.rmtree(tempdir)
+
+        .. testoutput::
+            :hide:
+            :options: +NORMALIZE_WHITESPACE
+
+            INFO: 'my_study' does not exist. Creating a new experiment
+
+    Args:
+        tracking_uri:
+            Set the tracking server URI.
+
+            - An empty string, or a local directory path, prefixed with ``file:``. Data is stored
+              locally in the provided directory (or ``./mlruns`` if empty).
+            - An HTTP URI like ``https://my-tracking-server:5000``.
+            - A Databricks workspace, provided as the string ``databricks`` or, to use a
+              Databricks CLI.
+              `profile <https://github.com/databricks/databricks-cli#installation>`_,
+              ``databricks://<profileName>``.
+        experiment:
+            Name of MLflow experiment to be activated. If not set ``study.study_name``
+            will be taken. Either ``experiment`` or ``study.study_name`` must be set.
+        metric_name:
+            Name of the metric. If not provided this will be called ``trial_value``.
+    """
+
+    def __init__(self, tracking_uri=None, experiment=None, metric_name=None):
+        # type: (str, str, str) -> None
```

Comment:
The arguments seems to be optional. Please replace `str` with `Optional[str]`.

Reply:
Good point. Changed this.